### PR TITLE
[MM-31360] downgrade error logs which doesn't break the flow

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We shouldn’t continue the operation by just logging the error. If we can continue, it means that the operation can still continue in the context so basically we can treat this as a warning/info/debug. If it is an actual error, we should just return the error and handle it. Also, added `Context.LogByCode(*model.AppError)` so that the errors don't caught by `web/handler` can also use the same rule.

PS: This PR does not include changes for `app`, `jobs` and `store` packages.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31360

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: services/awsmeter/awsmeter.go [hunk 1]
```diff
@@ -81,8 +81,7 @@ func newAWSMarketplaceMeteringService() (*marketplacemetering.MarketplaceMeterin
 
 	_, err = creds.Get()
 	if err != nil {
-		mlog.Error("session is invalid", mlog.String("error", err.Error()))
-		return nil, errors.New("cannot obtain credentials")
+		return nil, errors.Wrap(err, "cannot obtain credentials")
```
Comment (isacikgoz) [inline services/awsmeter/awsmeter.go:84]: This already logged in the call site.

File: services/telemetry/telemetry.go [hunk 2]
```diff
@@ -1075,42 +1075,42 @@ func (ts *TelemetryService) trackElasticsearch() {
 func (ts *TelemetryService) trackGroups() {
 	groupCount, err := ts.dbStore.Group().GroupCount()
 	if err != nil {
-		mlog.Error(err.Error())
+		mlog.Debug("Could not get group_count", mlog.Err(err))
```
Comment (isacikgoz) [inline services/telemetry/telemetry.go:1078]: I think errors for telemetries probably do not interest the system administrator.

File: web/context.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ import (
 
 type Context struct {
 	App           app.AppIface
-	Log           *mlog.Logger
+	Logger        *mlog.Logger
```
Comment (streamer45) [inline web/context.go:21]: Thank you :+1:

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

File: web/context.go [hunk 4]
```diff
@@ -85,14 +85,33 @@ func (c *Context) LogAuditWithUserId(userId, extraInfo string) {
 	}
 }
 
+func (c *Context) LogByCode(err *model.AppError) {
```
Comment (agnivade) [inline web/context.go:88]: Let's rename it to `LogErrorByCode` to make it clear what does it log.

Additionally, I think we should refactor out the actual log line printing from all the 3 methods to a separate internal method. This should keep all the fields that get printed in a single place. That method should only take the level and the apperror as its arguments.

**Review by agnivade — CHANGES_REQUESTED** [review]
Nice work. Suggested a small refactoring.

File: web/context.go (hunk 4 above)
Comment (isacikgoz) [reply web/context.go]: Updated, in fact removed all other 3 methods and now we can have one method for all cases.

File: cmd/mattermost/commands/server.go [hunk 5]
```diff
@@ -79,7 +79,7 @@ func runServer(configStore *config.Store, usedPlatform bool, interruptChan chan
 	defer server.Shutdown()
 
 	if usedPlatform {
-		mlog.Error("The platform binary has been deprecated, please switch to using the mattermost binary.")
+		mlog.Warn("The platform binary has been deprecated, please switch to using the mattermost binary.")
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply cmd/mattermost/commands/server.go:82]: Agreed. It doesn't make any harm though but still can cause a confusion. I will ask.

**Review by agnivade — APPROVED** [review]
Looks amazing

**Pull request merged**